vip-helpers/vip-liveblog.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/**
+ * Purge the Varnish cache for a liveblog on each new entry.
+ *
+ * Ensures that a Liveblog page isn't cached with stale mete data during an
+ * active liveblog.
+ *
+ * @param  int $comment_id ID of the comment for this new entry.
+ * @param  int $post_id    ID for this liveblog post.
+ */
+function wpcom_vip_liveblog_purge_on_new_entries( $comment_id, $post_id ){
+
+	// Get the URL for this Liveblog post.
+	$permalink = get_permalink( absint( $post_id ) );
+	if ( ! $permalink ) {
+		return;
+	}
+
+	if ( ! function_exists( 'wpcom_vip_purge_edge_cache_for_url' ) ) {
```

Comment:
I would presume that this `function_exists()` check is less expensive than trying to do a `get_permalink()`, so I'd swap over these guard clauses, and avoid a potentially un-necessary database query.